test/e2e/knative_serving_test.go

```diff
@@ -291,14 +296,79 @@ func testUserPermissions(t *testing.T, paCtx *test.Context, editCtx *test.Contex
 
 func waitForRouteServingText(t *testing.T, caCtx *test.Context, routeDomain, expectedText string) {
 	t.Helper()
-	_, err := pkgTest.WaitForEndpointState(
+	if _, err := pkgTest.WaitForEndpointState(
 		&pkgTest.KubeClient{Kube: caCtx.Clients.Kube},
 		t.Logf,
 		routeDomain,
 		pkgTest.EventuallyMatchesBody(expectedText),
 		"WaitForRouteToServeText",
-		true)
-	if err != nil {
+		true); err != nil {
 		t.Fatalf("The Route at domain %s didn't serve the expected text \"%s\": %v", routeDomain, expectedText, err)
 	}
 }
+
+func resetProxy(t *testing.T, caCtx *test.Context, proxyValue, envName string) {
+	if err := test.UpdateGlobalProxy(caCtx, proxyValue); err != nil {
+		t.Fatal("Failed to update proxy", err)
+	}
+	t.Log("wait for controller to be ready after update")
+	test.WaitForControllerEnvironment(caCtx, knativeServing, envName)
+}
+
+func testKnativeServingForGlobalProxy(t *testing.T, caCtx *test.Context) {
+	test.CleanupOnInterrupt(t, func() { resetProxy(t, caCtx, "", proxyVerificationEnvName) })
+	defer resetProxy(t, caCtx, "", proxyVerificationEnvName)
+
+	t.Log("update global proxy with empty proxy value")
+	resetProxy(t, caCtx, "", proxyVerificationEnvName)
+
+	t.Log("deploy successfully knative service after proxy update")
+	if _, err := test.WithServiceReady(caCtx, proxyHelloworldServiceSuccess, testNamespace, image); err != nil {
+		t.Fatal("Knative Service not ready", err)
+	}
+
+	t.Log("update global proxy with proxy server")
+	resetProxy(t, caCtx, "http://1.2.4.5:8999", httpProxy)
+
+	t.Log("deploy knative service after proxy update")
+	test.WithServiceReady(caCtx, proxyHelloworldService, testNamespace, proxyImage)
+
+	svc, err := test.GetService(caCtx, proxyHelloworldService, testNamespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var proxyExist bool
+
+	for _, cond := range svc.Status.Conditions {
+		// After global proxy update every call goes through proxy server
+		// Here it give unable to pull image because it tries to connect to not running http server
+		if strings.Contains(cond.Message, "dial tcp 1.2.4.5:8999: i/o timeout") {
+			// use bool variable here because service status have more than one conditions and if none of the condition matches
+			// then assume knative service failed because of some other reason
+			proxyExist = true
+			break
+		}
+	}
+
+	if !proxyExist {
+		t.Fatal("Service not ready", err)
```

Comment:
This `err` will never be something else than `nil` cause the last time you set err is in line 336 where you also check it immediately for nil-ness. You'd rather want to log `svc.Status` here, I think.